### DISCUSSION TITLE
Optimizing gas boiler efficiency: Seeking advice on multi-room heating control with Smart Thermostats

### DISCUSSION THREAD

**Discussion by wiktor-z** [jmcollin78/versatile_thermostat#202] [Q&A]
Thanks for the time and effort you put into implementing this thermostat. Until now, I've been using https://github.com/ScratMan/HASmartThermostat and still use it to control one of the relays responsible for underfloor heating, and I'm quite satisfied with this solution. Unfortunately, I can't figure out a working setup for 5 TuYa TS0601/hy368 (_TZE200_cpmgn2cf) thermostatic heads with it, hence my interest in your project.

I ran a test for one room with one radiator and one TRV (climate device in HA), but in the next step, I'd like to implement a similar control for the remaining rooms. I'm particularly interested in the "power management" aspect, but I have a gas boiler at home, so my power issue is the opposite of electric heating. I'd like the ability to open as many heads as possible simultaneously so that the gas boiler can operate under optimal conditions (many heat emitters available at the same time, the need to heat a larger volume of water in the system, the possibility of saving on condensation, avoiding frequent boiler cycling due to water overheating in the system).

Is there a way in this project to synchronize the heating activation for multiple rooms at once while maintaining the ability to control the temperature for each room separately? Of course, I understand that once the target temperature for a single room is reached, some heads will start closing independently, but every minute the gas boiler is heating water for a larger system is a gain for me.

I'm looking forward to any guidance, but, of course, I don't expect a ready-made solution handed to me on a silver platter :)

**Comment by jmcollin78** [comment]
Hello @wiktor-z ,

> I'm particularly interested in the "power management" aspect

Power management is only for electrical device I think. You will be able to hack the system but I don't even see how to do that. The purpose is to stop electrical heaters when power measurement is over a threshold.

> Is there a way in this project to synchronize the heating activation for multiple rooms at once

There is no way to synchronize multiple VTherm with native VTherm functions but it is possible with an external automation thanks to the service provided. So you will have to implement it on your side.

Keep us in touch is you find something suitable, as this could be a generical need for others.

**Reply by wiktor-z** [reply]
Thank you for the feedback. Currently, I'm working on an automation that takes into account the state of a virtual sensor. This sensor will activate the gas boiler if at least two thermostats require heating or if one thermostat requires heating for more than a specified time. It's not a perfect solution as it may lead to cooling down rooms. I need to conduct some tests to determine the optimal delay time that allows heating for more than one room, as this approach may have its limitations.

Using https://github.com/ScratMan/HASmartThermostat potentially has the advantage that I could configure PWM to the same duration. This could potentially result in all thermostatic heads activating simultaneously. However, at the moment, I'm facing challenges configuring this add-on to work with my TRVs.

Thank you for your patience, and I'll keep you posted on the progress and any insights gained from testing.

**Reply by wiktor-z** [reply]
Unfortunately, in another threads ( https://github.com/ScratMan/HASmartThermostat/discussions/186#discussioncomment-7635551 https://github.com/ScratMan/HASmartThermostat/discussions/183#discussioncomment-7635621 ), the author explained in more detail the principle of operation and calculation of the PWM cycle start moment. So, for now, it seems that I won't be able to use this HASmartThermostat, at least not as the only tool for synchronizing heating activation.
